sleyster_a07b.js

```diff
@@ -70,35 +63,35 @@ console.log(groceryList);
 //Step 2
 
 //function buildFoodsList defined to fill the badFoos array with user inputed items
-function buildFoodsList (badFoods, ...args) {
+function buildFoodsList(foods, ...args) {
     for (arg of args) {
-        badFoods.unshift(arg);
+        foods.unshift(arg);
+        console.log("we are in the for loop");
     }
 }
 //function getPrompt defined to output prompt for user to input items or quit or view items
-function getPrompt (groceryList) {
-    console.log(`Your grocery list contains ${groceryList.length} items. 
+function getPrompt(grocery) {
+    console.log(`Your grocery list contains ${grocery.length} items. 

```

Comment:
this is where our undefined is coming from! you are logging from inside the function - but then in the main program - it tries to log whatever is returned from here - and it returns nothing! Undefined! I should have caught that.
Anyway - in this function, I'd have a variable that you build the prompt in and then return that var to be logged. Make sense?